src/Service.Tests/Unittests/ConfigFileWatcherUnitTests.cs

```diff
@@ -0,0 +1,150 @@
+    },
+    ""host"": {
+      ""cors"": {
+        ""origins"": [
+          ""http://localhost:5000""
+        ],
+        ""allow-credentials"": false
+      },
+      ""authentication"": {
+        ""provider"": ""StaticWebApps""
+      },
+      ""mode"": """ + initialMode + @"""
+    }
+  },
+  ""entities"": {}
+}";
+            string updatedConfig = @"
+{
+  ""$schema"": ""https://github.com/Azure/data-api-builder/releases/download/vmajor.minor.patch/dab.draft.schema.json"",
+  ""data-source"": {
+    ""database-type"": ""mssql"",
+    ""connection-string"": ""Server=test;Database=test;User ID=test"",
+    ""options"": {
+      ""set-session-context"": true
+    }
+  },
+  ""runtime"": {
+    ""rest"": {
+      ""enabled"": " + updatedRestEnabled.ToString().ToLower() + @",
+      ""path"": """ + updatedRestPath + @"""
+    },
+    ""graphql"": {
+      ""enabled"": " + updatedGQLEnabled.ToString().ToLower() + @",
+      ""path"": """ + updatedGQLPath + @""",
+      ""allow-introspection"": " + updatedGQLIntrospection.ToString().ToLower() + @"
+    },
+    ""host"": {
+      ""cors"": {
+        ""origins"": [
+          ""http://localhost:5000""
+        ],
+        ""allow-credentials"": false
+      },
+      ""authentication"": {
+        ""provider"": ""StaticWebApps""
+      },
+      ""mode"": """ + updatedMode + @"""
+    }
+  },
+  ""entities"": {}
+}";
+            string configName = "hotreload-config.json";
+            if (File.Exists(configName))
+            {
+                File.Delete(configName);
+            }
+
+            FileSystem fileSystem = new();
+            fileSystem.File.WriteAllText(configName, initialConfig);
+            FileSystemRuntimeConfigLoader configLoader = new(fileSystem, configName, string.Empty);
+            RuntimeConfigProvider configProvider = new(configLoader);
+            // Must GetConfig() to start file watching
+            RuntimeConfig runtimeConfig = configProvider.GetConfig();
+            // assert we have a valid config
+            Assert.IsNotNull(runtimeConfig);
+            Assert.AreEqual(initialRestEnabled, runtimeConfig.Runtime.Rest.Enabled);
+            Assert.AreEqual(initialRestPath, runtimeConfig.Runtime.Rest.Path);
+            Assert.AreEqual(initialGQLEnabled, runtimeConfig.Runtime.GraphQL.Enabled);
+            Assert.AreEqual(initialGQLPath, runtimeConfig.Runtime.GraphQL.Path);
+            Assert.AreEqual(initialGQLIntrospection, runtimeConfig.Runtime.GraphQL.AllowIntrospection);
+            Assert.AreEqual(initialMode, runtimeConfig.Runtime.Host.Mode);
+            // Simulate change to the config file
+            fileSystem.File.WriteAllText(configName, updatedConfig);
+            // Give file watcher enough time to see the change
+            System.Threading.Thread.Sleep(1000);
+            // Hot Reloaded config
+            runtimeConfig = configProvider.GetConfig();
```

Comment:
I think it makes more sense to have this coverage as part of the integration testing. This unit test is really just for testing the unit of code that is doing the actual hot reloading itself. In this case we'd just be testing the same unit of code twice but it would be triggered from two different locations. With integration testing we can cover all of these unique code paths and make sure that they all work together. The integration tests depend on covering some additional scenarios that will be in with future work (multiple config files in particular).

Reply:
sounds good.